uaclient/tests/test_cli.py

```diff
@@ -828,208 +829,54 @@ def test_status_human_readable_warning(
 
 
 class TestSetupLogging:
-    @pytest.mark.parametrize("level", (logging.INFO, logging.ERROR))
-    @mock.patch("uaclient.cli.util.we_are_currently_root", return_value=False)
-    @mock.patch("uaclient.log.get_user_log_file")
-    def test_console_log_configured_if_not_present(
+    def test_correct_handlers_added_to_logger(
         self,
-        m_get_user,
-        m_we_are_currently_root,
-        level,
-        capsys,
         logging_sandbox,
         FakeConfig,
-        tmpdir,
     ):
-        m_get_user.return_value = tmpdir.join("user.log").strpath
-        with mock.patch(
-            "uaclient.cli.config.UAConfig", return_value=FakeConfig()
-        ):
-            setup_logging(level, logging.INFO)
-        logging.log(level, "after setup")
-        logging.log(level - 1, "not present")
-
-        _, err = capsys.readouterr()
-        assert "after setup" in err
-        assert "not present" not in err
+        console_level = logging.INFO
+        log_level = logging.DEBUG
+        logger = logging.getLogger("logger_a")
 
-    @mock.patch("uaclient.cli.util.we_are_currently_root", return_value=False)
-    @mock.patch("uaclient.log.get_user_log_file")
-    def test_console_log_configured_if_already_present(
-        self,
-        m_get_user,
-        m_we_are_currently_root,
-        capsys,
-        logging_sandbox,
-        FakeConfig,
-        tmpdir,
-    ):
-        m_get_user.return_value = tmpdir.join("user.log").strpath
-        logging.getLogger().addHandler(logging.StreamHandler(sys.stderr))
+        handler = logging.StreamHandler(sys.stderr)
+        handler.setLevel(logging.ERROR)
+        handler.set_name("ua-test-console")
+        logger.addHandler(handler)
 
         with mock.patch(
             "uaclient.cli.config.UAConfig", return_value=FakeConfig()
         ):
-            logging.error("before setup")
-            setup_logging(logging.INFO, logging.INFO)
-            logging.error("after setup")
-
-        # 'before setup' will be in stderr, so check that setup_logging
-        # configures the format
-        _, err = capsys.readouterr()
-        assert "ERROR: before setup" not in err
-        assert "ERROR: after setup" in err
+            setup_logging(console_level, log_level, logger=logger)
+        assert len(logger.handlers) == 2
+        for idx, handler in enumerate(logger.handlers):
+            if idx == 0:
+                assert handler.name == "ua-console"
+                assert handler.level == console_level
+            elif idx == 1:
+                assert handler.name == "ua-file"
+                assert handler.level == log_level
```

Comment:
IMO this is a bit easier to read
```suggestion
        assert logger.handlers[0].name == "ua-console"
        assert logger.handlers[0].level == console_level
        assert logger.handlers[1].name == "ua-file"
        assert logger.handlers[1].level == log_level
```

Reply:
I had implemented this way, and though it was a bit manual.

Reply:
It does feel more manual, but there is no need for anything more complex. I don't think that feeling is a bad thing.

IMO the goal of tests is to minimize abstraction and do things directly as possible. I think the overhead of reading a for loop and figuring out that that it asserts something differently for two different indexes makes the test harder to read. (aside: more generally, I consider any loop or if-statement to be a code-smell in a test)

In both versions of this code, there is an assumption that there are only two items in the list (as asserted on the line above). I think this assumption is more obvious in the manual version.